### PR TITLE
Improve plugins installation

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: neo-cli/CLI/MainService.Plugins.cs [hunk 1]
```diff
@@ -17,51 +18,48 @@ partial class MainService
...
             }
-            finally
+            using (response)
             {
-                if (isTemp)
+                using Stream stream = response.GetResponseStream();
+                using ZipArchive zip = new(stream, ZipArchiveMode.Read);
+                try
+                {
+                    zip.ExtractToDirectory(".");
```
Comment (shargon) [inline neo-cli/CLI/MainService.Plugins.cs:55]: The release will contain the folder name?

Comment (erikzhang) [reply neo-cli/CLI/MainService.Plugins.cs:55]: Yes.

**Review by shargon — DISMISSED** [review]
@superboyiii tested?

File: neo-cli/CLI/MainService.Plugins.cs [hunk 2]
```diff
@@ -17,51 +18,48 @@ partial class MainService
...
+                    .Select(p => new
+                    {
+                        Version = Version.Parse(p["tag_name"].GetString().TrimStart('v')),
+                        Assets = p["assets"].GetArray()
+                    })
+                    .OrderByDescending(p => p.Version)
+                    .First(p => p.Version <= version_core).Assets
+                    .First(p => p["name"].GetString() == $"{pluginName}.zip");
+                request = WebRequest.CreateHttp(asset["browser_download_url"].GetString());
+                response = (HttpWebResponse)request.GetResponse();
```
Comment (superboyiii) [inline neo-cli/CLI/MainService.Plugins.cs:47]:
```suggestion
                try
                {
                    JObject asset = releases.GetArray()
                    .Where(p => !p["tag_name"].GetString().Contains('-'))
                    .Select(p => new
                    {
                        Version = Version.Parse(p["tag_name"].GetString().TrimStart('v')),
                        Assets = p["assets"].GetArray()
                    })
                    .OrderByDescending(p => p.Version)
                    .First(p => p.Version <= version_core).Assets
                    .First(p => p["name"].GetString() == $"{pluginName}.zip");
                    request = WebRequest.CreateHttp(asset["browser_download_url"].GetString());
                    response = (HttpWebResponse)request.GetResponse();
                }
                catch (Exception) { throw new Exception("Plugin not existed."); }
```

File: neo-cli/CLI/MainService.Plugins.cs [hunk 3]
```diff
@@ -17,51 +18,52 @@ partial class MainService
...
+                        Version = Version.Parse(p["tag_name"].GetString().TrimStart('v')),
+                        Assets = p["assets"].GetArray()
+                    })
+                    .OrderByDescending(p => p.Version)
+                    .First(p => p.Version <= version_core).Assets
+                    .First(p => p["name"].GetString() == $"{pluginName}.zip");
+                    request = WebRequest.CreateHttp(asset["browser_download_url"].GetString());
+                    response = (HttpWebResponse)request.GetResponse();
+                }
+                catch (Exception) { throw new Exception("Plugin is not existed."); }
```
Comment (shargon) [inline neo-cli/CLI/MainService.Plugins.cs:51]:
```suggestion
                catch (Exception) { throw new Exception("Plugin doesn't exist."); }
```

**Review by superboyiii — APPROVED** [review]
Test OK

**Review by superboyiii — APPROVED** [review]

**Comment by superboyiii** [comment]
Test OK

**Pull request merged**